### PR TITLE
Merge Nulecule specification into Atomic App

### PR DESCRIPTION
**Pull request by cdrage**
This commit merges the documentation located at
github.com/projectatomic/nulecule into Atomic App.

The changes included with the merge are:
* Remove reference to developer tooling being added
* Remove reference to example folder, now directs
to nulecule-library
* Remove redundant information (links to #nulecule
irc channel) already present in root README.md
* Move .json spec files to json folder
* Clean up documentation links to link to the
correct directory
* Capitalize spec doc files, move them all to a
single folder
* Add link to NULECULE_FILE.md reference in main
spec README.md
* Add link to specification in the root README.md
* Update description of Nulecule in the main README.md

### CODE REVIEW

**Comment by cdrage** [comment]
Fork can be found here: https://github.com/cdrage/atomicapp/tree/merge-spec-to-atomicapp

With the inclusion of the spec into Atomic App, we will also copy-over any current open issues with the Nulecule spec and mark with the `specification` label.

**Comment by cdrage** [comment]
@rtnpro @Preeticp @dustymabe @kadel 
any feedback would be great for mergin' :)

File: README.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ This README contains some high level overview information on Atomic App. The det
 8. [Specification coverage](docs/spec_coverage.md)
```
Comment (dustymabe) [inline README.md]: Should we merge docs/nulecule.md into docs/spec/ somewhere?

**Comment by dustymabe** [comment]
One comment. otherwise LGTM - what should we do with the Nulecule repo?

File: README.md (hunk 1 above)
Comment (cdrage) [inline README.md]: @dustymabe considering that the Nulecule format in Atomic App is _still_ different than the format in the spec, I would say to keep it separate. For example, external files for docker source'd applications (ex. the mariadb) is specified in the spec, but we do not support it yet in atomic app

Comment (dustymabe) [inline README.md]: should we rip out anything in the spec that isn't supported by atomicapp and then list those features in a TODO list somewhere. One of the benefits of putting these two together would be to reduce confusion IMHO. Having them out of sync in the same repo wouldn't make much sense.

Comment (cdrage) [inline README.md]: hence the reaosn for spec_coverage.md doc.

Yeah, we should rip those out... but that's a spec change that should be put in a separate PR.

**Comment by cdrage** [comment]
@dustymabe 

have it forward to atomicapp OR add a deprecation notice / this has moved.

Once this PR has been 100% merged, I'll start creating the new issues related to the Nulecule spec in here.

File: README.md (hunk 1 above)
Comment (dustymabe) [inline README.md]: ok - as long as the plan isn't to leave them out of sync in the same repo.

**Comment by dustymabe** [comment]
thanks LGTM
